src/WebJobs.Script.WebHost/Diagnostics/FunctionsVersionLoggerService.cs

```diff
@@ -0,0 +1,128 @@
+using Microsoft.Extensions.Options;
+using Newtonsoft.Json;
+
+namespace Microsoft.Azure.WebJobs.Script.Diagnostics
+{
+    /// <summary>
+    /// Service responsible for logging versions
+    /// </summary>
+    public class FunctionsVersionLoggerService : IHostedService, IDisposable
+    {
+        private const double IntervalInSecondsDev = 1;    // interval for dev unit testing (1 s) in which the logs would be dumped
+        private readonly ILogger _logger;
+        private readonly Timer _timer;
+        private readonly TimeSpan _interval;
+        private readonly IEnvironment _environment;
+        private bool _disposed;
+        private double intervalInSecondsProd = 3600;      // default interval (1 hour) in which the logs would be dumped
+
+        public FunctionsVersionLoggerService(ILogger<FunctionsVersionLoggerService> logger, IEnvironment environment)
+        {
+            _timer = new Timer(OnTimer, null, Timeout.Infinite, Timeout.Infinite);
+            _environment = environment ?? throw new ArgumentNullException(nameof(environment));
+            _interval = TimeSpan.FromSeconds(this.Interval);
+            _logger = logger;
+        }
+
+        public virtual double Interval
+        {
+            get { return (_environment.GetEnvironmentVariable(EnvironmentSettingNames.AzureWebsiteSlotName) == "development") ? IntervalInSecondsDev : intervalInSecondsProd; }
+            set { intervalInSecondsProd = value; }
+        }
+
+        public Task StartAsync(CancellationToken cancellationToken)
+        {
+            // start the timer by setting the due time
+            SetTimerInterval((int)_interval.TotalMilliseconds);
+
+            return Task.CompletedTask;
+        }
+
+        public Task StopAsync(CancellationToken cancellationToken)
+        {
+            // stop the timer if it has been started
+            _timer?.Change(Timeout.Infinite, Timeout.Infinite);
+
+            return Task.CompletedTask;
+        }
+
+        private void OnTimer(object state)
+        {
+            PublishLogsSamples();
+        }
+
+        private void PublishLogs()
+        {
+            try
+            {
+                _logger.LogInformation(
+                    "FunctionsExtensionVersion = {FunctionsExtensionVersion},  Framework = {Framework}, FrameworkVersion = {FrameworkVersion}, SlotName = {SlotName}",
+                    _environment.GetEnvironmentVariable(EnvironmentSettingNames.FunctionsExtensionVersion),
+                    _environment.GetEnvironmentVariable(EnvironmentSettingNames.Framework),
+                    _environment.GetEnvironmentVariable(EnvironmentSettingNames.FrameworkVersion),
+                    _environment.GetEnvironmentVariable(EnvironmentSettingNames.AzureWebsiteSlotName));
+            }
+            catch (Exception exc) when (!exc.IsFatal())
+            {
+                throw;
+            }
+        }
+
+        private void PublishLogsSamples()
+        {
+            try
```

Comment:
`PublishLogs` is already within a `try`/`catch` block.  Is there a reason why we have this additional method here? Can we just move the error logging there and get rid of `PublishLogsSamples`?

Reply:
Done